tactics/class_tactics.ml

```diff
@@ -661,154 +732,214 @@ module Search = struct
     let sigma = Goal.sigma gl in
     let unique = not info.search_dep || is_unique env sigma concl in
     let backtrack = needs_backtrack env sigma unique concl in
-    if !typeclasses_debug > 0 then
-      Feedback.msg_debug
-        (pr_depth info.search_depth ++ str": looking for " ++
-           Printer.pr_econstr_env (Goal.env gl) sigma concl ++
-           (if backtrack then str" with backtracking"
-            else str" without backtracking"));
-    let secvars = compute_secvars gl in
-    let poss =
-      e_possible_resolve hints info.search_hints secvars info.search_only_classes env sigma concl in
-    (* If no goal depends on the solution of this one or the
+    if !typeclasses_caching &&
+         TypeclassCache.exists
```

Comment:
I do not know why I used `exists` instead of `mem` :) Anyway, switching to `mem` sped up things a bit and now with cache, the compilation of my benchmark it is 3:02 (compared to 4:55 without)!